2017/ru/0xa1-injection.md

```diff
@@ -0,0 +1,77 @@
+# A1:2017 Внедрение
+
+| Источники угроз/Векторы атак | Недостатки безопасности           | Последствия               |
+| -- | -- | -- |
+| Зависит от прил. : Сложность эксплуатации 3 | Распространенность 2 : Сложность обнаружения 3 | Технические 3 : Для бизнеса ?|
+| Почти любой источник данных может оказаться вектором для внедрения: переменные окружения, параметры, внешние и внутренние веб-службы, а также все типы пользователей. [Внедрения](https://www.owasp.org/index.php/Injection_Flaws) становятся возможными, если злоумышленник может отправлять интерпретатору вредоносные данные. | Внедрения особенно распространены в старом коде. Уязвимости часто встречаются в SQL- LDAP-, XPath- или NoSQL-запросах, системных командах, XML-обработчиках, SMTP-заголовках, языках выражений и ORM-запросах. Внедрения легко обнаружить при анализе кода. Сканеры и фаззеры могут помочь злоумышленникам найти подобные уязвимости. |Внедрения могут привести к потере данных, их повреждению или разглашению третьим лицам, а также к отказу в обслуживании. В некоторых случаях контроль над узлом может быть полностью перехвачен. Последствия для бизнеса зависят от критичности приложения и данных.|
+
+
+## Является ли приложение уязвимым?
+
+Приложение уязвимо, если:
+
+* вводимые пользователем данные не проверяются, не фильтруются или не очищаются;
+* динамические запросы или непараметризованные вызовы без контекстного экранирования напрямую используются в интерпретаторе;  
+* вредоносные данные используются в поисковых параметрах объектно-реляционного отображения для извлечения дополнительной, критичной информации;
+* вредоносные данные используются или добавляются т.о., что SQL-код или команды содержат структурные и вредоносные данные в динамических запросах, командах или хранимых процедурах.
+
+Наиболее распространенными являются SQL-, NoSQL-, ORM-, LDAP-, EL- или OGNL-внедрения, а также внедрения команд ОС. То же самое касается всех интерпретаторов. Анализ исходного кода является лучшим способом обнаружения внедрений, за которым следует полное автоматизированное тестирование всех вводимых параметров, заголовков, URL, куки, JSON-, SOAP- и XML-данных. Организации также могут включать в процесс непрерывной интеграции и развертывания ПО (CI/CD) статическое ([SAST](https://www.owasp.org/index.php/Source_Code_Analysis_Tools)) и динамическое ([DAST](https://www.owasp.org/index.php/Category:Vulnerability_Scanning_Tools)) тестирование кода и приложений для обнаружения новых уязвимостей перед внедрением приложений в производство.
+<details>
+    <summary>
+        <b><i><font size="2" color="aqua">ORM | LDAP | EL | OGNL</font></i></b>
+    </summary>
+    <font size="2" color="aquamarine">
+        <b>ORM</b> → Объектно-реляционное отображение <br>
+        <b>LDAP</b> → Облегченный протокол доступа к каталогам<br>
+        <b>EL</b> → Язык выражений<br>
+        <b>OGNL</b> → Объектно-графовый язык навигации
+    </font>
+</details>
```

Comment:
Does it make sense to use html formatting with colored fonts here? Moreover, aquamarine color does not contrast with white background.

Can we replace it in the following way?

```markdown
### ORM | LDAP | EL | OGNL

* **ORM** - Объектно-реляционное отображение
* **LDAP** - Облегченный протокол доступа к каталогам
* **EL** - Язык выражений
* **OGNL** - Объектно-графовый язык навигации
```

It makes sense to use `details` section for long texts, but not in this case.